api/document/serializers/content.py

```diff
@@ -164,9 +182,9 @@ class BaseAnnotationSerializer(serializers.Serializer):
     inlines = InlinesField(is_leaf_node=False)
     text = TextField(is_leaf_node=False)
 
-    @property
-    def CONTENT_TYPE(self):     # noqa; this is a constant
-        raise NotImplementedError('Unknown annotation type')
+    # Subclasses need to implement these.
+    CONTENT_TYPE: Optional[str] = None
+    ANNOTATION_CLASS: Optional[Type['Annotation']] = None
```

Comment:
Yep, I believe it generally does play well with it, and I was originally thinking of using it but then I realized that I didn't know how well it'd play together with multiple inheritance, since our class also has to inherit from the serializer... I could try that out though.

Reply:
So the good news is that [mypy pays attention to abstract methods and properties](http://mypy.readthedocs.io/en/latest/class_basics.html#abstract-base-classes-and-multiple-inheritance) even if they don't have `ABCMeta` as a metaclass, which solves the potential multiple inheritance issues.

However, the bad news is that I'm still running into the same problem that I ran into even when I tried making `CONTENT_TYPE` a non-abstract class property, which is that mypy mis-infers the property as a `Callable[[], str]` instead of an actual `str`, for some reason.  I think maybe it only supports _instance_ properties or something... What's also odd is that I can't find any mention of this kind of thing via web searches. Hmm.

Anywys, I am just going to merge this for now and we can revisit it later if we figure out a better solution.

Reply:
Oh wait, I just realized that I could simply instantiate the class in `register`, which solves all the problems, so I will just do that.